packages/pigeon/platform_tests/test_plugin/android/src/main/kotlin/com/example/test_plugin/ProxyApiTestApiImpls.kt

```diff
@@ -0,0 +1,680 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package com.example.test_plugin
+
+import io.flutter.plugin.common.BinaryMessenger
+
+class ProxyApiTestClass : ProxyApiSuperClass(), ProxyApiInterface
+
+open class ProxyApiSuperClass
+
+interface ProxyApiInterface
+
+class ProxyApiRegistrar(binaryMessenger: BinaryMessenger) :
+    PigeonProxyApiRegistrar(binaryMessenger) {
+  override fun getPigeonApiProxyApiTestClass(): PigeonApiProxyApiTestClass {
+    return ProxyApiTestClassApi(this)
+  }
+
+  override fun getPigeonApiProxyApiSuperClass(): PigeonApiProxyApiSuperClass {
+    return ProxyApiSuperClassApi(this)
+  }
+}
+
+class ProxyApiTestClassApi(pigeonRegistrar: ProxyApiRegistrar) :
+    PigeonApiProxyApiTestClass(pigeonRegistrar) {
+
+  override fun pigeon_defaultConstructor(
+      aBool: Boolean,
+      anInt: Long,
+      aDouble: Double,
+      aString: String,
+      aUint8List: ByteArray,
+      aList: List<Any?>,
+      aMap: Map<String?, Any?>,
+      anEnum: ProxyApiTestEnum,
+      aProxyApi: ProxyApiSuperClass,
+      aNullableBool: Boolean?,
+      aNullableInt: Long?,
+      aNullableDouble: Double?,
+      aNullableString: String?,
+      aNullableUint8List: ByteArray?,
+      aNullableList: List<Any?>?,
+      aNullableMap: Map<String?, Any?>?,
+      aNullableEnum: ProxyApiTestEnum?,
+      aNullableProxyApi: ProxyApiSuperClass?,
+      boolParam: Boolean,
+      intParam: Long,
+      doubleParam: Double,
+      stringParam: String,
+      aUint8ListParam: ByteArray,
+      listParam: List<Any?>,
+      mapParam: Map<String?, Any?>,
+      enumParam: ProxyApiTestEnum,
+      proxyApiParam: ProxyApiSuperClass,
+      nullableBoolParam: Boolean?,
+      nullableIntParam: Long?,
+      nullableDoubleParam: Double?,
+      nullableStringParam: String?,
+      nullableUint8ListParam: ByteArray?,
+      nullableListParam: List<Any?>?,
+      nullableMapParam: Map<String?, Any?>?,
+      nullableEnumParam: ProxyApiTestEnum?,
+      nullableProxyApiParam: ProxyApiSuperClass?
+  ): ProxyApiTestClass {
+    return ProxyApiTestClass()
+  }
+
+  override fun attachedField(pigeon_instance: ProxyApiTestClass): ProxyApiSuperClass {
+    return ProxyApiSuperClass()
+  }
+
+  override fun staticAttachedField(): ProxyApiSuperClass {
+    return ProxyApiSuperClass()
+  }
+
+  override fun aBool(pigeon_instance: ProxyApiTestClass): Boolean {
```

Comment:
Is there a reason these are using hard-code values instead of being echo methods like the existing tests? Echo allows for more test coverage. E.g., being able to test that a nullable-return method can successfully return both a null and a non-null value.

Reply:
Why was this change necessary?

Reply:
It made the feature mentioned in another comment possible: https://github.com/flutter/packages/pull/6371#discussion_r1536177357

Essentially, there wasn't a way for me to retrieve the constructor values in the `ProxyApiBaseCodec` for a class if the original `dartDefaultConstructor` was used.